### PR TITLE
Deployment Guide for Wildfly Servers

### PR DESCRIPTION
**Pull request by rsoika**
See discussion here: https://github.com/mvc-spec/ozark/issues/123

Please check the page /ozark/index.md if it is as you expected it. I am not sure if the link is correct. I was not able get build running on my Debian system.

### CODE REVIEW

**Comment by Daniel-Dos** [comment]
Hi @rsoika 

For me that ok.

File: src/ozark/install_wildfly.md [hunk 1]
```diff
@@ -0,0 +1,91 @@
+---
+title: Deployment Guide for Wildfly
+template: page.html
+---
+
+## Deployment Guide for Wildfly
```
Comment (chkal) [inline src/ozark/install_wildfly.md]: Could you move the file into a subdirectory `docs`. That would be great.

In this case the link in `index.md` should be:

    [Deployment Guide for Wildfly](docs/install_wildfly.html)

File: src/ozark/index.md [hunk 2]
```diff
@@ -12,6 +12,11 @@ Ozark is the reference implementation (RI) for JSR 371.
   * [Source Code](https://github.com/mvc-spec/ozark)
   * [Issue Tracker](https://github.com/mvc-spec/ozark/issues)
   
+### Documentation
+
+  * [Deployment Guide for Wildfly](install_wildfly)
```
Comment (chkal) [inline src/ozark/index.md]: You need to add `.html` to the file name. So it should be `[Deployment Guide for Wildfly](install_wildfly.html)`

**Comment by rsoika** [comment]
I think I need to install a newer version of npn to run the build on my debian system. In the Debian Jessie packages 'npm' is actually provided in version  1.4.21

This is the error message I got during the build:

```
rsoika@ralpus-ultrabook:~/git/www.mvc-spec.org$ npm run build

> www.mvc-spec.org@1.0.0 build /home/rsoika/git/www.mvc-spec.org
> node build.js

sh: 1: node: not found

npm ERR! www.mvc-spec.org@1.0.0 build: `node build.js`
npm ERR! Exit status 127
npm ERR! 
npm ERR! Failed at the www.mvc-spec.org@1.0.0 build script.
npm ERR! This is most likely a problem with the www.mvc-spec.org package,
npm ERR! not with npm itself.
npm ERR! Tell the author that this fails on your system:
npm ERR!     node build.js
npm ERR! You can get their info via:
npm ERR!     npm owner ls www.mvc-spec.org
npm ERR! There is likely additional logging output above.
npm ERR! System Linux 4.2.0-0.bpo.1-amd64
npm ERR! command "/usr/bin/nodejs" "/usr/bin/npm" "run" "build"
npm ERR! cwd /home/rsoika/git/www.mvc-spec.org
npm ERR! node -v v0.10.29
npm ERR! npm -v 1.4.21
npm ERR! code ELIFECYCLE
npm WARN This failure might be due to the use of legacy binary "node"
npm WARN For further explanations, please read
/usr/share/doc/nodejs/README.Debian
 
npm ERR! 
npm ERR! Additional logging details can be found in:
npm ERR!     /home/rsoika/git/www.mvc-spec.org/npm-debug.log
npm ERR! not ok code 0
```

Because I am not working with npm at all, I did not make any additional installation. I hoe my changes now are working.

===
Ralph

**Comment by chkal** [comment]
Thanks a lot! Looks good to me. I'll merge this now and it should appear on the page in a few minutes.

Did you install both `node` and `npm`? From the error it looks like the `node` binary cannot be found or isn't on the path. I guess the `npm` version doesn't matter so much. But you will need a recent version of `node`. But this may be another problem.

**Comment by chkal** [comment]
BTW: Please see [my recent post](https://groups.google.com/forum/#!topic/ozark-dev/KqSwfstr_1I) on the ozark-dev list about my work on getting rid of the Jersey dependency in our code.

**Comment by rsoika** [comment]
The problem with my node installation was indeed a conflict with a debian package called 'node'. In debian 'nodejs' must be installed instead. I fixed it. Now I am running node version v0.10.29

```
>node --version
v0.10.29
```

Is this version sufficient? 
Running now the build command leads to the following message:

```
> www.mvc-spec.org@1.0.0 build /home/rsoika/git/www.mvc-spec.org
> node build.js


/home/rsoika/git/www.mvc-spec.org/lib/metalsmith/news.js:11
  return (files, metalsmith, done) => {
                                    ^
SyntaxError: Unexpected token >
    at Module._compile (module.js:439:25)
    at Object.gnodeJsExtensionCompiler [as .js] (/home/rsoika/git/www.mvc-spec.org/node_modules/metalsmith/node_modules/gnode/index.js:61:10)
    at Module.load (module.js:356:32)
    at Function.Module._load (module.js:312:12)
    at Module.require (module.js:364:17)
    at require (module.js:380:17)
    at Object.<anonymous> (/home/rsoika/git/www.mvc-spec.org/build.js:7:12)
    at Module._compile (module.js:456:26)
    at Object.Module._extensions..js (module.js:474:10)
    at Module.load (module.js:356:32)
    at Function.Module._load (module.js:312:12)
    at Function.Module.runMain (module.js:497:10)
    at startup (node.js:119:16)
    at node.js:906:3

npm ERR! www.mvc-spec.org@1.0.0 build: `node build.js`
npm ERR! Exit status 8
npm ERR! 
npm ERR! Failed at the www.mvc-spec.org@1.0.0 build script.
npm ERR! This is most likely a problem with the www.mvc-spec.org package,
npm ERR! not with npm itself.
npm ERR! Tell the author that this fails on your system:
npm ERR!     node build.js
npm ERR! You can get their info via:
npm ERR!     npm owner ls www.mvc-spec.org
npm ERR! There is likely additional logging output above.
npm ERR! System Linux 4.2.0-0.bpo.1-amd64
npm ERR! command "/usr/bin/nodejs" "/usr/bin/npm" "run" "build"
npm ERR! cwd /home/rsoika/git/www.mvc-spec.org
npm ERR! node -v v0.10.29
npm ERR! npm -v 1.4.21
npm ERR! code ELIFECYCLE
npm WARN This failure might be due to the use of legacy binary "node"
npm WARN For further explanations, please read
/usr/share/doc/nodejs/README.Debian
 
npm ERR! 
npm ERR! Additional logging details can be found in:
npm ERR!     /home/rsoika/git/www.mvc-spec.org/npm-debug.log
npm ERR! not ok code 0
```

I fear I need a newer version of node ?

**Comment by chkal** [comment]
Well, `v0.10.29` is from 2014. So it is really really old. 

I'm using Fedora but I always use the official package repository by the nodejs guys to get the latest stable release. This will See all the details here:

https://nodejs.org/en/download/package-manager/#debian-and-ubuntu-based-linux-distributions

**Comment by rsoika** [comment]
ok - so now it's clear ;-) 
sometimes debian is really outdated. I will upgrade my installation...

**Pull request merged**